awx/playbooks/project_update.yml

```diff
@@ -224,28 +225,45 @@
         - "{{ project_path | quote }}/roles/requirements.yaml"
         - "{{ project_path | quote }}/roles/requirements.yml"
       changed_when: "'was installed successfully' in galaxy_result.stdout"
-      environment: "{{ galaxy_task_env }}"
+      environment: "{{ galaxy_task_env | combine(additional_galaxy_env) }}"
       when: roles_enabled | bool
       tags:
         - install_roles
 
     - name: Fetch galaxy collections from collections/requirements.(yml/yaml)
       ansible.builtin.command: >
         ansible-galaxy collection install -r {{ item }}
-        --collections-path {{ projects_root }}/.__awx_cache/{{ local_path }}/stage/requirements_collections
         {{ ' -' + 'v' * ansible_verbosity if ansible_verbosity else '' }}
       args:
         chdir: "{{ project_path | quote }}"
       register: galaxy_collection_result
       with_fileglob:
         - "{{ project_path | quote }}/collections/requirements.yaml"
         - "{{ project_path | quote }}/collections/requirements.yml"
-        - "{{ project_path | quote }}/requirements.yaml"
-        - "{{ project_path | quote }}/requirements.yml"
       changed_when: "'Installing ' in galaxy_collection_result.stdout"
-      environment: "{{ additional_collections_env | combine(galaxy_task_env) }}"
+      environment: "{{ galaxy_task_env | combine(additional_galaxy_env) }}"
       when:
         - "ansible_version.full is version_compare('2.9', '>=')"
         - collections_enabled | bool
       tags:
         - install_collections
+
+    - name: Fetch galaxy roles and collections from requirements.(yml/yaml)
+      ansible.builtin.command: >
+        ansible-galaxy install -r {{ item }}
+        {{ ' -' + 'v' * ansible_verbosity if ansible_verbosity else '' }}
+      args:
+        chdir: "{{ project_path | quote }}"
+      register: galaxy_combined_result
+      with_fileglob:
+        - "{{ project_path | quote }}/requirements.yaml"
+        - "{{ project_path | quote }}/requirements.yml"
+      changed_when: "'Nothing to do.' not in galaxy_combined_result.stdout"
```

Comment:
Could your "Nothing to do" criteria also be applied to the other 2 Galaxy commands?

Also, `environment:` may be shared between them all, maybe we could set those with a `block:`

Reply:
Looks like we could do collections:
```
ansible-galaxy collection install awx.awx
Starting galaxy collection install process
Nothing to do. All requested collections are already installed. If you want to reinstall them, consider using `--force`.
```

But not the role install:
```
ansible-galaxy role install geerlingguy.jenkins
Starting galaxy role install process
[WARNING]: - geerlingguy.jenkins (4.3.0) is already installed - use --force to change version to unspecified
```

In addition to extracting environment I also made a var to shorten up the command line.